rainier-core/src/main/scala/com/stripe/rainier/compute/Translator.scala

```diff
@@ -148,7 +177,20 @@ private class Translator {
         () =>
           binaryExpr(toExpr(x), Const(a.toDouble), ring.times)
     }
-    combineTree(lazyExpr, ring)
+  }
+
+  private def combineTerms(terms: Seq[(Real, BigDecimal)], ring: Ring): Expr = {
+    val lazyExprs = makeLazyExprs(terms, ring)
+    combineTree(lazyExprs, ring)
+  }
+
+  private def combineSumTerms(terms: Seq[(Real, BigDecimal)],
```

Comment:
seems a bit weird to pass the ring in here when it's always going to be the multiplyRing